src/Apps/Order/Routes/Payment/PaymentContext/OrderPaymentContext.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useReducer, useContext, createContext, Dispatch } from "react"
+import { CommercePaymentMethodEnum } from "__generated__/Payment_order.graphql"
+import { orderPaymentReducer } from "./orderPaymentReducer"
+import createLogger from "Utils/logger"
+
+const logger = createLogger("[dev: OrderPaymentContext] state:")
+
+/**
+ * Actions
+ */
+
+enum OrderPaymentActions {
+  SET_SELECTED_BANK_ACCOUNT_ID = "SET_SELECTED_BANK_ACCOUNT_ID",
+  SET_SELECTED_PAYMENT_METHOD = "SET_SELECTED_PAYMENT_METHOD",
+}
+
+type OrderPaymentActionsPayload = {
+  [OrderPaymentActions.SET_SELECTED_BANK_ACCOUNT_ID]: string
+  [OrderPaymentActions.SET_SELECTED_PAYMENT_METHOD]: CommercePaymentMethodEnum
+}
+
+type OrderPaymentAction = ActionMap<OrderPaymentActionsPayload>[keyof ActionMap<
+  OrderPaymentActionsPayload
+>]
+
+type ActionMap<M extends { [index: string]: any }> = {
+  [Key in keyof M]: M[Key] extends undefined
+    ? {
+        type: Key
+      }
+    : {
+        type: Key
+        payload: M[Key]
+      }
+}
+
+/**
+ * State
+ */
+
+type OrderPaymentState = {
+  selectedBankAccountId: string
+  selectedPaymentMethod: CommercePaymentMethodEnum | string
+}
+
+const initialOrderPaymentState = {
+  selectedBankAccountId: "",
+  selectedPaymentMethod: "",
+}
+
+/**
+ * Context
+ */
+
+const OrderPaymentContext = createContext<{
+  state: OrderPaymentState
+  dispatch: Dispatch<OrderPaymentAction>
+}>({
+  state: initialOrderPaymentState,
+  dispatch: () => null,
+})
+
+/**
+ * Provider
+ */
+
+const OrderPaymentContextProvider: React.FC = ({ children }) => {
+  const [state, dispatch] = useReducer(
+    orderPaymentReducer,
+    initialOrderPaymentState
+  )
+
+  // for developers' convenience only
+  if (process.env.NODE_ENV === "development") {
+    logger.log(state)
+  }
+
+  return (
+    <OrderPaymentContext.Provider value={{ state, dispatch }}>
+      {children}
+    </OrderPaymentContext.Provider>
+  )
+}
+
+/**
+ * Hook
+ */
+
+const useOrderPaymentContext = () => {
+  const { state, dispatch } = useContext(OrderPaymentContext)
+  return { state, dispatch }
```

Comment:
As I mentioned at the knowledge share we can return an action creator instead of dispatch.
![Screenshot from 2022-09-09 14-49-32](https://user-images.githubusercontent.com/79979820/189343564-833527fc-ade1-4570-a856-d45873d95460.png)
And then use:
![Screenshot from 2022-09-09 14-51-36](https://user-images.githubusercontent.com/79979820/189343778-41189449-b81f-42fb-8451-2841e4189796.png)

This example uses the hook as a place where we can create an function which will create an action and pass the action to dispatch. Then the function can be used at the needed place.

Reply:
Yup, @AlicanAkyuz - this is basically the same thing I was saying in my comment up above! Then everything is very readable by just using autocomplete / intellisense, without us having to wire up dispatches and actions and so on.